docs/docsite/rst/dev_guide/testing/sanity/no-underscore-variable.rst

```diff
@@ -0,0 +1,28 @@
+Sanity Tests » no-underscore-variable
+=====================================
+
+In the future, Ansible may use the identifier ``_`` to internationalize its
+message strings.  To be ready for that, we need to make sure that there are
+no conflicting identifiers defined in the code base.
+
+In common practice, ``_`` is frequently used as a dummy variable (a variable
+to receive a value from a function where the value is useless and never used).
+In Ansible, we're using the identifier ``dummy`` for this purpose instead.
```

Comment:
ah, looking again it wasn't pylint showing the error, it was flake8.  (I have both turned on for my checkers)

```
diff --git lib/ansible/parsing/vault/__init__.py lib/ansible/parsing/vault/__init__.py
index 9040cbf..47a066a 100644
--- lib/ansible/parsing/vault/__init__.py
+++ lib/ansible/parsing/vault/__init__.py
@@ -526,7 +526,7 @@ class VaultLib:
         # the known vault secrets.
         if not C.DEFAULT_VAULT_ID_MATCH:
             # Add all of the known vault_ids as candidates for decrypting a vault.
-            vault_id_matchers.extend([_vault_id for _vault_id, _dummy in self.secrets if _vault_id != vault_id])
+            vault_id_matchers.extend([_vault_id for _vault_id, dummy in self.secrets if _vault_id != vault_id])
 
         matched_secrets = match_secrets(self.secrets, vault_id_matchers)
 
[newswoop:F25:ansible (vault_pylint %)]$ flake8  lib/ansible/parsing/vault/__init__.py 
lib/ansible/parsing/vault/__init__.py:529:64: F812 list comprehension redefines 'dummy' from line 490
```